src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -0,0 +1,554 @@
+    val resultDFFieldNames = rawDF.schema.fieldNames
+    if (!resultDFFieldNames.contains(apiMeta.dataframeColumn) && apiMeta.dataframeColumn != "*") { // if known API but return column doesn't exist
+      val asDFErrMsg = s"The API endpoint is not returning the " +
+        s"expected structure, column ${apiMeta.dataframeColumn} is expected and is not present in the dataframe.\nIf this module " +
+        s"references data that does not exist or to which the Overwatch account does have access, please remove the " +
+        s"scope. For example: if you have 'pools' scope enabled but there are no pools or Overwatch PAT doesn't " +
+        s"have access to any pools, this scope must be removed."
+      logger.log(Level.ERROR, asDFErrMsg)
+      throw new Exception(asDFErrMsg)
+    } else if (apiMeta.dataframeColumn == "*") { //Selecting all of the column.
+      rawDF
+    }
+    else { //Selecting specific columns as per the metadata.
+      rawDF.select(explode(col(apiMeta.dataframeColumn)).alias(apiMeta.dataframeColumn)).select(col(apiMeta.dataframeColumn + ".*"))
+    }
+  }
+
+  /**
+   * Converting the API response to Dataframe.
+   *
+   * @return Dataframe which is created from the API response.
+   */
+  def asDF(): DataFrame = {
+    var apiResultDF: DataFrame = null;
+    if (_apiResponseArray.size == 0 && !apiMeta.storeInTempLocation) { //If response contains no Data.
+      val errMsg = s"API CALL Resulting DF is empty BUT no errors detected, progressing module. " +
+        s"Details Below:\n$buildGenericErrorMessage"
+      throw new ApiCallEmptyResponse(errMsg, true)
+    } else if (_apiResponseArray.size != 0 && !apiMeta.storeInTempLocation) { //If API response don't have pagination/volume of response is not huge then we directly convert the response which is in-memory to spark DF.
+      apiResultDF = spark.read.json(Seq(_apiResponseArray.toString).toDS())
+    } else if (apiMeta.storeInTempLocation) { //Read the response from the Temp location/Disk and convert it to Dataframe.
+      apiResultDF = spark.read.json(successTempPath)
+
+    }
+    extrapolateSupportedStructure(apiResultDF)
+  }
+
+
+  def executeMultiThread(): util.ArrayList[String] = {
+    try {
+      val response = getResponse
+      responseCodeHandler(response)
+      _apiResponseArray.add(response.body)
+      if (apiMeta.storeInTempLocation) {
+        if (apiEnv.successBatchSize <= _apiResponseArray.size()) { //Checking if its right time to write the batches into persistent storage
+          val responseFlag = Helpers.writeMicroBatchToTempLocation(successTempPath, _apiResponseArray.toString)
+          if (responseFlag) { //Clearing the resultArray in-case of successful write
+            setApiResponseArray(new util.ArrayList[String]())
+          }
+        }
+      }
+      paginate(response.body)
+      if (_printFinalStatusFlag) {
+        println(buildSleepDetailMessage())
```

Comment:
only print inside the hibernate functions -- not necessary when not hibernating. This is duplicate

Reply:
actually the name of the function is confusing so renaming the function as buildDetailMessage(). 
At end of the API call this function will print the stats of the API call like.

API call Endpoint: https://adb-2222170229861029.9.azuredatabricks.net/api/2.0/clusters/resize
queryString: {"cluster_id": "0511-074343-c456edtd","num_workers":2}
Server Busy Count:0
Total Sleep Time:0 seconds
Successful api response count:1